SH4RE/SH4RE/CreateListing/CreateListingView.swift

```diff
@@ -210,57 +220,112 @@ struct CreateListingView: View {
+            imgPath = "listingimages/" + editListing.id + "/" + String(index) + ".jpg"
+            arrayImgs.append(imgPath)
+            storageManager.upload(image: pic, path: imgPath)
+            index += 1
+        }
+        if (documentUpdate(collectionPath: "Listings", documentID: editListing.id, data: ["image_path" : arrayImgs])) {
+            NSLog("error");
+        }
+        showPostAlertX = true
+    }
+    func validatePost () -> Bool {
         if (title.isEmpty || cost.isEmpty || postalCode.isEmpty ||
             pictures.isEmpty || categorySelection.isEmpty || description.isEmpty ||
             (availabilitySelection.isEmpty && availabilityCalendar.selectedDates.isEmpty)) {
             errorInField = true
+            return false
         }
-        if (!errorInField) {
-            // upload data fields
-            var calAvail = [Any]()
-            if (!availabilityCalendar.selectedDates.isEmpty) {
-                for date in availabilityCalendar.selectedDates {
-                    calAvail.append(date)
-                }
-            }
-            else {
-                calAvail.append(availabilitySelection)
-            }
-            let listingFields = ["Title": title, "Description" : description, "Price" : cost, "Category" : categorySelection, "Availability": calAvail, "Address": postalCode, "UID": getCurrentUserUid()] as [String : Any]
-            let documentID = documentWrite(collectionPath: "Listings", data: listingFields)
-            
-            // upload images and add paths to data fields
-            var index = 1
-            var imgPath = ""
-            var arrayImgs:[String] = []
-            for pic in pictures {
-                imgPath = "listingimages/" + documentID + "/" + String(index) + ".jpg"
-                arrayImgs.append(imgPath)
-                storageManager.upload(image: pic, path: imgPath)
-                index += 1
-            }
-            if (documentUpdate(collectionPath: "Listings", documentID: documentID, data: ["image_path" : arrayImgs])) {
-                NSLog("error");
-            }
-            showPostAlertX = true
-
-            //reset inputs
-            resetInputs()
-        }
+        return true
     }
     
     var body: some View {
         ZStack {
             Color.backgroundGrey.ignoresSafeArea()
+                .onAppear() {
+                    isEditing = editListing.title != ""
+                    if (isEditing) {
+                        title = editListing.title
+                        description = editListing.description
+                        cost = editListing.price
+                        postalCode = editListing.address
+                        availabilityCalendar.selectedDates = editListing.availability
+                        imagesCount = editListing.imagepath.count
+                        categorySelection = editListing.category
+                        imagesCount = (imagesCount <  5) ? imagesCount + 1 : imagesCount
```

Comment:
what does this do?

Reply:
adjusts image count so we can display the add image button if there are less than 5 images